tests/integration/components/a11y-landmark-test.js

```diff
@@ -1,16 +1,100 @@
 import { moduleForComponent, test } from 'ember-qunit';
 import hbs from 'htmlbars-inline-precompile';
+import LANDMARK_NAVIGATION_ROLES from 'ember-a11y-landmarks/utils/landmark-navigation-roles';
+import set from 'ember-metal/set';
+import run from 'ember-runloop';
 
 moduleForComponent('a11y-landmark', 'Integration | Component | a11y landmark', {
   integration: true
 });
 
-test('it renders', function(assert) {
+test('throws error by default when no tagName or landmarkRole specified', function(assert) {
+  const expectedErrorMessage = 'Assertion Failed: Must specify either tagName or landmarkRole';
+
+  assert.expectAssertion(() => {
+    this.render(hbs`
+      {{#a11y-landmark}}
+        default landmark
+      {{/a11y-landmark}}
+    `);
+  }, expectedErrorMessage);
+});
+
+test('search form', function(assert) {
   this.render(hbs`
-    {{#a11y-landmark tagName="form"}}
-      template block text
+    {{#a11y-landmark tagName='form' landmarkRole='search'}}
+      search form
     {{/a11y-landmark}}
   `);
 
-  assert.equal(this.$().text().trim(), 'template block text');
+  assert.equal(this.$('form').text().trim(), 'search form', 'renders form');
+  assert.equal(this.$('form').attr('role'), 'search', 'landmark role is search');
+});
+
+test('sets role for each landmark', function(assert) {
+  const landmarks = Object.keys(LANDMARK_NAVIGATION_ROLES);
+
+  assert.expect(landmarks.length);
+
+  landmarks.forEach((landmark) => {
+    run(() => { set(this, 'tagName', landmark); });
+
+    this.render(hbs`
+      {{#a11y-landmark tagName=tagName}}
```

Comment:
I don't have the time to dig more into this right now but the Travis builds for the older LTS versions seem to be failing because of this. Apparently, setting the component's `tagName` params to a variable sends in an `[Object object]` (haven't figured out what that looks like yet). Using a literal string like `tagName="nav"` seems to work fine.

Would love to know what's wrong / what the fix is if someone knows. :)